src/sentry/buffer/base.py

```diff
@@ -53,7 +53,7 @@ def incr(self, model, columns, filters, extra=None):
             }
         )
 
-    def process_pending(self):
+    def process_pending(self, *args, **kwargs):
```

Comment:
Seems like `shard` should just become part of the standard API, even if implementations choose not to respect it, rather than defining this method signature as literally any parameters.

Reply:
Yeah, I did that, then figured `shard` only made sense to redis, and didn't wanna dictate whatever future API this might be for now.

Reply:
I guess the main task itself is acknowleding shard as well, so I guess it does make sense to just keep this in base. I'll do that.